src/OpenConext/UserLifecycle/Domain/Service/ProgressReporterInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\UserLifecycle\Domain\Service;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+interface ProgressReporterInterface
```

Comment:
Not 100% sure this should be in the domain layer. What is your motivation for adding it here? 

My rationale is: reporting on the progress of the batch deprovision command is not really important to the domain and fits better in the Application layer.

Reply:
We've now created a new interface for the console-specific parts.